drivers/led/sn32/rgb_matrix_sn32f24xb.c

```diff
@@ -182,6 +233,21 @@ static void update_pwm_channels(PWMDriver *pwmp) {
 static void rgb_callback(PWMDriver *pwmp) {
     // Disable the interrupt
     pwmDisablePeriodicNotification(pwmp);
+#if (SN32_PWM_CONTROL == SOFTWARE)
+    for (uint8_t col_idx = 0; col_idx < LED_MATRIX_COLS; col_idx++) {
+        uint8_t led_col = chan_col_order[col_idx];
+        if ((pwmp->period <= (led_col_duty_cycle[led_col])) && (led_col_duty_cycle[led_col] > 0)) {
```

Comment:
`        if ((pwmp->period <= (led_col_duty_cycle[led_col])) && (led_col_duty_cycle[led_col] > 0)) {`
needs to be 
`        if (((uint8_t)(pwmp->ct->TC)<= (led_col_duty_cycle[led_col])) && (led_col_duty_cycle[led_col] > 0)) {`
for it to be somewhat correct. More debugging needed